src/Utils/request/query.ts

```diff
@@ -2,18 +2,40 @@ import careConfig from "@careConfig";
 
 import { QueryError } from "@/Utils/request/queryError";
 import { getResponseBody } from "@/Utils/request/request";
-import { QueryOptions, Route } from "@/Utils/request/types";
+import { Route } from "@/Utils/request/types";
+// Remove conflicting QueryOptions import
 import { makeHeaders, makeUrl } from "@/Utils/request/utils";
 
+// Define QueryOptions locally, including the 'headers' property
+export interface QueryOptions<TBody> {
+  body?: TBody;
+  queryParams?: Record<string, any>;
+  pathParams?: Record<string, any>;
+  silent?: boolean;
+  signal?: AbortSignal;
+  headers?: Record<string, string>; // Add headers support
+}
+
 async function queryRequest<TData, TBody>(
   { path, method, noAuth }: Route<TData, TBody>,
   options?: QueryOptions<TBody>,
 ): Promise<TData> {
   const url = `${careConfig.apiUrl}${makeUrl(path, options?.queryParams, options?.pathParams)}`;
 
+  // Convert Headers object to a plain object
+  const defaultHeaders = Object.fromEntries(
+    makeHeaders(noAuth ?? false).entries(),
+  );
+
+  // Merge default headers with custom headers, with custom headers taking precedence
+  const headers = {
+    ...defaultHeaders,
+    ...(options?.headers || {}),
+  };
+
```

Comment:
I guess you may be confused about the objective. This would "overwrite" the headers. But the intended behaviour is to "append" the headers while preserving the previously set headers.

Refer: https://developer.mozilla.org/en-US/docs/Web/API/Headers/append

<img width="792" alt="image" src="https://github.com/user-attachments/assets/76ad8928-8353-4a2f-84ac-2a440fa53a96" />

Reply:
I am sorry about that, I'll fix this and let you know